haproxy/tests/conftest.py

```diff
@@ -5,25 +5,77 @@
 import time
 import logging
 import mock
-import tempfile
-import shutil
 import getpass
 
-from . import common
+from copy import deepcopy
+
+from datadog_checks.dev import TempDir, WaitFor, docker_run
+from datadog_checks.dev.utils import create_file, file_exists
+from datadog_checks.haproxy import HAProxy
+
+from .common import HERE, CHECK_CONFIG, USERNAME, PASSWORD, STATS_URL, STATS_URL_OPEN
 
 log = logging.getLogger('test_haproxy')
 
 
+def wait_for_haproxy():
+    res = requests.get(STATS_URL, auth=(USERNAME, PASSWORD))
+    res.raise_for_status()
+
+    res_open = requests.get(STATS_URL_OPEN)
+    res_open.raise_for_status()
+
+
+@pytest.fixture(scope='session')
+def dd_environment():
+    env = {}
+    with TempDir() as temp_dir:
+        host_socket_path = os.path.join(temp_dir, 'datadog-haproxy-stats.sock')
+
+        env['HAPROXY_CONFIG_DIR'] = os.path.join(HERE, 'compose')
+        env['HAPROXY_CONFIG'] = os.path.join(HERE, 'compose', 'haproxy.cfg')
+        if os.environ.get('HAPROXY_VERSION', '1.5.11').split('.')[:2] >= ['1', '7']:
+            env['HAPROXY_CONFIG'] = os.path.join(HERE, 'compose', 'haproxy-1_7.cfg')
+        env['HAPROXY_CONFIG_OPEN'] = os.path.join(HERE, 'compose', 'haproxy-open.cfg')
+        env['HAPROXY_SOCKET_DIR'] = temp_dir
+
+        with docker_run(
```

Comment:
Let's do https://github.com/DataDog/integrations-core/blob/bd17c03c87212dd4f6863a00a93914bc11402af7/mcache/tests/conftest.py#L35 and mark tests too so more people can run tests for this check